core/src/main/scala/caliban/GraphQL.scala

```diff
@@ -78,7 +78,10 @@ trait GraphQL[-R] { self =>
    * Fails with a [[caliban.CalibanError.ValidationError]] if the schema is invalid.
    */
   final def interpreter(implicit trace: Trace): IO[ValidationError, GraphQLInterpreter[R, CalibanError]] =
-    validateRootSchema.map { schema =>
+    ZIO.fromEither(cachedInterpreter)
```

Comment:
I'm thinking whether this should initialise the interpreter upfront or lazily, e.g.,

```scala
val i = cachedInterpreter
ZIO.fromEither(i)
```

Thoughts?

Reply:
Makes sense yeah